File: src/Software Engineer Prepare Kit/12/solution.go
package main

type SinglyLinkedListNode struct {
	data int32
	next *SinglyLinkedListNode
}

type SinglyLinkedList struct {
	head *SinglyLinkedListNode
	tail *SinglyLinkedListNode
}

func (singlyLinkedList *SinglyLinkedList) InsertNodeIntoSinglyLinkedList(nodeData int32) {
	node := &SinglyLinkedListNode{
		next: nil,
		data: nodeData,
	}

	if singlyLinkedList.head == nil {
		singlyLinkedList.head = node
	} else {
		singlyLinkedList.tail.next = node
	}

	singlyLinkedList.tail = node
}

func DeleteDuplicates(head *SinglyLinkedListNode) *SinglyLinkedListNode {
	_head := head

	for _head != nil && _head.next != nil {
		next := _head.next

		for next != nil && _head.data == next.data {
			next = next.next
		}

		_head.next = next
		_head = _head.next
	}

	return head
}

File: src/Software Engineer Prepare Kit/12/solution.ts
class SinglyLinkedListNode {
	data: number
	next: SinglyLinkedListNode | null

	constructor(nodeData: number) {
		this.data = nodeData
		this.next = null
	}
}

class SinglyLinkedList {
	head: SinglyLinkedListNode | null
	tail: SinglyLinkedListNode | null

	constructor() {
		this.head = null
		this.tail = null
	}

	insertNode(nodeData: number): void {
		const node = new SinglyLinkedListNode(nodeData)

		if (this.head == null) {
			this.head = node
		} else {
			this.tail!.next = node
		}

		this.tail = node
	}
}

export default function deleteDuplicates(head: SinglyLinkedListNode): SinglyLinkedListNode {
	let _head: SinglyLinkedListNode | null = head

	while (_head && _head.next) {
		let next: SinglyLinkedListNode | null = _head.next
		while (next && _head.data == next.data) next = next.next

		_head.next = next
		_head = _head.next
	}

	return head
}

export { SinglyLinkedList }
